geode-core/src/main/java/org/apache/geode/cache/client/internal/PutOp.java

```diff
@@ -299,13 +300,39 @@ protected Object processResponse(Message msg, Connection con) throws Exception {
           VersionTag tag = (VersionTag) msg.getPart(partIdx).getObject();
           // we use the client's ID since we apparently don't track the server's ID in connections
           tag.replaceNullIDs((InternalDistributedMember) con.getEndpoint().getMemberId());
-          event.setVersionTag(tag);
+          checkForDeltaConflictAndSetVersionTag(tag, con);
         }
         return oldValue;
       }
       return null;
     }
 
+    void checkForDeltaConflictAndSetVersionTag(VersionTag versionTag, Connection connection)
+        throws Exception {
+      RegionEntry regionEntry = ((EntryEventImpl) event).getRegionEntry();
+      if (regionEntry == null) {
+        event.setVersionTag(versionTag);
+        return;
+      }
+      VersionStamp versionStamp = regionEntry.getVersionStamp();
+      if (deltaSent && versionTag.getEntryVersion() != versionStamp.getEntryVersion() + 1) {
```

Comment:
It should be ">" not "!=", in the case of "<", you should not call the get.

Reply:
I am not sure if client delta put will generate region version less than current local cache version. However, use ">" might be more clear in this case.